compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -183,22 +186,32 @@ namespace cppmicroservices
             auto registry = mgr->GetRegistry();
             auto logger = mgr->GetLogger();
             auto configNotifier = mgr->GetConfigNotifier();
-            auto managers = mgr->GetManagers();
-
             try
             {
                 auto compManager = std::make_shared<ComponentManagerImpl>(newMetadata,
                                                                           registry,
                                                                           bundle.GetBundleContext(),
                                                                           logger,
                                                                           asyncWorkService,
-                                                                          configNotifier,
-                                                                          managers);
+                                                                          configNotifier);
                 if (registry->AddComponentManager(compManager))
                 {
-                    managers->push_back(compManager);
-                    compManager->Initialize();
-                }
+                    auto extension = extensionRegistry->Find(bundle.GetBundleId());
+                    if (extension)
+                    {
+                        extension->AddComponentManager(compManager);
+                        compManager->Initialize();
```

Comment:
what should happen if `extension->AddComponentManager(compManager);` fails? should `compManager->Initialize();` still be called?

Reply:
No but the only way extension->AddComponentManager can fail is if it throws an exception. Since this is in a try block, control will jump to the catch.